### PR TITLE
Dynamically generate message class in JS

### PR DESCRIPTION
**Pull request by kenny-y**
Use the spec from parser to dynamically generate ROS message class in JS, assemble it with exactly the same memory layout with C++. Now it works with the first message: std_msgs/msg/String.msg

P.S. please use [squash & merge] button when merging...

### CODE REVIEW

**Comment by kenny-y** [comment]
With this dynamic-message-generating mechanism, when developers add or modify a message, s/he won't need to re-install (or compile) this rclnodejs module/addon.

File: example/publisher-example.js [hunk 1]
```diff
@@ -12,29 +12,41 @@
...
 // limitations under the License.
 
+// Usage: lanuch ./install/bin/examples_rclcpp_minimal_subscriber_lambda
+//  to display the message published by this program
+
 'use strict';
 
+const StructType = require('ref-struct');
+const ref = require('ref');
+const ffi = require('ffi');
```
Comment (minggangw) [inline example/publisher-example.js]: these 3 lines above are useless

File: rosidl_gen/message.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+'use strict';
+
+const fs = require('fs.promised');
+const mkdirp = require('mkdirp');
+const path = require('path');
+const dot = require('dot');
+const requireFromString = require('require-from-string');
+
+const primitiveTypes = require('../rosidl_gen/generator_primitive.js');
+const Parser = require('../rosidl_parser/rosidl_parser.js');
```
Comment (minggangw) [inline rosidl_gen/message.js]: not capital here

File: rosidl_gen/message.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+const message = {
+  createMessage: function(messageType) {
+    const rosInstallPath = process.env.AMENT_PREFIX_PATH;
+    const packagePath = rosInstallPath + '/share/' +
+      messageType.pkgName + '/' + messageType.msgSubfolder + '/' + messageType.msgName + '.msg';
+
+    return new Promise(function(resolve, reject) {
+      Parser.parseMessageFile(messageType.pkgName, packagePath).then((spec) => {
```
Comment (minggangw) [inline rosidl_gen/message.js]: ditto

**Review by minggangw — APPROVED** [review]
LGTM, and some small problems, thanks!

**Pull request merged**